spec/views/sign_up/completions/show.html.erb_spec.rb

```diff
@@ -90,4 +90,31 @@
       expect(rendered).to include('9**-**-***6')
     end
   end
+
+  context 'MFA CTA banner' do
+    let(:multiple_factors_enabled) { false }
+
+    before do
+      allow(IdentityConfig.store).to receive(:select_multiple_mfa_options).and_return(true)
+      @multiple_factors_enabled = multiple_factors_enabled
+    end
+
+    it 'shows a banner if the user selects one MFA option' do
+      render
+      expect(rendered).to have_content(t('mfa.second_method_warning.text'))
+    end
+  end
+
+  context 'shows the banner with multiple factors enabled' do
+    let(:multiple_factors_enabled) { true }
+
+    before do
+      allow(IdentityConfig.store).to receive(:select_multiple_mfa_options).and_return(false)
+    end
+
+    it 'does not show a banner' do
+      render
+      expect(rendered).not_to have_content(t('mfa.second_method_warning.text'))
+    end
+  end
```

Comment:
I think both of these separate `context` blocks should be combined, since they're both testing the same "MFA CTA banner" behavior. And since they're not combined currently, I don't think the `let` block on line 109 would be doing anything, since it's assigned only on line 99 (in a separate context).

Maybe something like...

```suggestion

  describe 'MFA CTA banner' do
    let(:multiple_factors_enabled) { nil }
    let(:select_multiple_mfa_options) { nil }

    before do
      allow(IdentityConfig.store).to receive(:select_multiple_mfa_options).
        and_return(select_multiple_mfa_options)
      @multiple_factors_enabled = multiple_factors_enabled
    end

    context 'with multiple factors disabled' do
      let(:multiple_factors_enabled) { false }

      context 'with multiple MFA feature flag disabled' do
        let(:select_multiple_mfa_options) { false }

        it 'does not show a banner' do
          render
          expect(rendered).not_to have_content(t('mfa.second_method_warning.text'))
        end
      end

      context 'with multiple MFA feature flag enabled' do
        let(:select_multiple_mfa_options) { true }

        it 'shows a banner if the user selects one MFA option' do
          render
          expect(rendered).to have_content(t('mfa.second_method_warning.text'))
        end
      end
    end

    context 'with multiple factors enabled' do
      let(:multiple_factors_enabled) { true }

      context 'with multiple MFA feature flag disabled' do
        let(:select_multiple_mfa_options) { false }

        it 'does not show a banner' do
          render
          expect(rendered).not_to have_content(t('mfa.second_method_warning.text'))
        end
      end

      context 'with multiple MFA feature flag enabled' do
        let(:select_multiple_mfa_options) { true }

        it 'does not show a banner' do
          render
          expect(rendered).not_to have_content(t('mfa.second_method_warning.text'))
        end
      end
    end
  end
```

Reply:
Combining these is a good idea. I think it will also be helpful to test that url to make sure it redirects to the two factor options page.